spring-app/src/main/java/pl/edu/agh/cqm/service/PingServiceImpl.java

```diff
@@ -34,14 +33,77 @@
     public void doMeasurement() {
         for (String domain : cqmConfiguration.getCdns()) {
             try {
-                rttSampleRepository.save(ping(domain));
+                String type = cqmConfiguration.getActiveTestsType();
+                if (type.equals("ICMP")) {
+                    rttSampleRepository.save(pingICMP(domain, type));
+                } else if (type.equals("TCP")) {
+                    rttSampleRepository.save(pingTCP(domain, type));
+                }
             } catch (IOException e) {
                 logger.error(e.getMessage());
             }
         }
     }
 
-    private RTTSample ping(String host) throws IOException {
+    private RTTSample pingTCP(String host, String type) throws IOException {
+        DecimalFormat format = (DecimalFormat) DecimalFormat.getInstance();
+        DecimalFormatSymbols symbols = format.getDecimalFormatSymbols();
+        char sep = symbols.getDecimalSeparator();
+
+        String command = String.join(" ",
+                "nping --tcp --delay ", "0" + sep + "2", "-c", cqmConfiguration.getActiveTestsIntensity() + "", host);
+        logger.info("Starting active sampling with command \"" + command + "\"");
+        BufferedReader inputStream = runSystemCommand(command);
+        // reading output stream of the command
+        List<String> lines = inputStream.lines().collect(Collectors.toList());
+
+        double std = 0;
+        double[] vals = new double[2*5]; //new ArrayList<>();
+        double[] stds = new double[5]; //new ArrayList<>();
+        for (int i = 2; i < lines.size()-3 ; i++) {
+            String line = lines.get(i);
+            double val = getValFromString(line.replaceAll("s", "qazwsx"), "((\\d+)(\\.)(\\d+))qazwsx");
+            if (val != -1) {
+                vals[i-2] = val;
+            }
+        }
+        for (int i = 0; i < (int)(vals.length/2) ; i++) {
```

Comment:
is type cast necessary here?

Reply:
there shouldn't be space before that semicolon. Use code formatter from intellij.